frontend/packages/console-app/src/components/terminal/__tests__/MinimizeRestoreButton.spec.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Button } from '@patternfly/react-core';
+import { OutlinedWindowMinimizeIcon, OutlinedWindowRestoreIcon } from '@patternfly/react-icons';
+import MinimizeRestoreButton from '../MinimizeRestoreButton';
+
+describe('MinimizeRestoreButton', () => {
+  it('should render a button', () => {
+    const wrapper = shallow(
+      <MinimizeRestoreButton minimizeText="Minimize" restoreText="Restore" onClick={() => null} />,
+    );
+    expect(wrapper.find(Button).exists()).toEqual(true);
+  });
+
+  it('should render minimize button when minimized is true', () => {
+    const wrapper = shallow(
+      <MinimizeRestoreButton
+        minimizeText="Minimize"
+        restoreText="Restore"
+        minimize
+        onClick={() => null}
+      />,
+    );
+    expect(wrapper.find(OutlinedWindowMinimizeIcon).exists()).toEqual(true);
+  });
+
+  it('should render restore button when minimized is false', () => {
+    const wrapper = shallow(
+      <MinimizeRestoreButton
+        minimizeText="Minimize"
+        restoreText="Restore"
+        minimize={false}
+        onClick={() => null}
+      />,
+    );
+    expect(wrapper.find(OutlinedWindowRestoreIcon).exists()).toEqual(true);
+  });
+
+  it('should invoke onclose callback with argument true when minimized button clicked and false when restore button clicked', () => {
+    const onClose = jest.fn();
+    const wrapper = shallow(
+      <MinimizeRestoreButton
+        minimizeText="Minimize"
+        restoreText="Restore"
+        minimize
+        onClick={onClose}
+      />,
+    );
+    expect(wrapper.find(OutlinedWindowMinimizeIcon).exists()).toEqual(true);
+    expect(wrapper.find(OutlinedWindowRestoreIcon).exists()).toEqual(false);
+    // click on minimize button
+    wrapper.find(Button).simulate('click');
+    expect(onClose).toHaveBeenCalledWith(true);
+    wrapper.setProps({ minimize: false });
+    expect(wrapper.find(OutlinedWindowMinimizeIcon).exists()).toEqual(false);
+    expect(wrapper.find(OutlinedWindowRestoreIcon).exists()).toEqual(true);
+    // click on restore button
+    wrapper.find(Button).simulate('click');
+    expect(onClose).toHaveBeenCalledWith(false);
```

Comment:
Did you test the inverse of this? It passes as well. I think the logic of this under the hood is "did it ever get called with <value>"... so if I flip this to `true` it passes because the previous invocation did pass `true`. If you say make it `null` it errors out because it was never called with `null`. I believe I provided two links in my last comment in the hopes of more or less providing documentation. Sorry -- probably could have been more explicit.